File: exercises/floor02/tests/test_ex01_booleans.py
import importlib.util
import sys
from pathlib import Path

EX = Path(__file__).resolve().parents[1] / "ex01_booleans.py"

spec = importlib.util.spec_from_file_location("ex01_booleans", EX)
mod = importlib.util.module_from_spec(spec)
sys.modules["ex01_booleans"] = mod
spec.loader.exec_module(mod)


def test_is_hurt():
    assert mod.is_hurt is True, "7 < 30 -> hurt"


def test_is_critical():
    assert mod.is_critical is True, "7 <= 30 * 0.25 (7.5) -> critical"


def test_can_open_door():
    assert mod.can_open_door is True, "has_key and door_locked"


def test_should_retreat():
    assert mod.should_retreat is True, "critical and no potions -> run"


def test_in_darkness():
    assert mod.in_darkness is True, "not torch_lit and depth > 1"


def test_fatal():
    assert mod.fatal is False, "hp is 7 -> not fatal. The robust check: player_hp <= 0"

File: exercises/cpp/floor02/tests/test_ex01_booleans.cpp
#include "../../devtest.h"
#include "../ex01_booleans.cpp"

int main() {
    CHECK_EQ(is_hurt(), true, "7 < 30 -> hurt");
    CHECK_EQ(is_critical(), true, "7 <= 30*0.25 (7.5) -> critical");
    CHECK_EQ(can_open_door(), true, "has_key && door_locked");
    CHECK_EQ(should_retreat(), true, "critical && potions == 0");
    CHECK_EQ(in_darkness(), true, "!torch_lit && depth > 1");
    CHECK_EQ(fatal(), false, "hp is 7 -> not fatal; the check is hp <= 0");
    return dt_summary();
}
